tests/suites/test_suite_pkparse.function

```diff
@@ -114,26 +114,18 @@ exit:
 /* END_CASE */
 
 /* BEGIN_CASE */
-void pk_parse_key( char *key_data, char *result_str, int result )
+void pk_parse_key( char *key_data, int result )
 {
     mbedtls_pk_context pk;
-    unsigned char buf[2000];
-    unsigned char output[2000];
+    unsigned char buf[128];
```

Comment:
This is uncomfortably small. We may well add larger test data without thinking about it. The old test framework in 2.7 isn't convenient for this, but please at least add a check that `data_len <= sizeof(buf)` after the call to `unhexify`, which will have a decent chance of catching a buffer overflow.

Reply:
Good point, I added a test for overflow.